x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/ConfigInitialNode.java

```diff
@@ -172,68 +158,67 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
-                return; // silent error because we wish the node to start as usual (skip auto config) if config dir is not writable
-            }
+            throw new UserException(ExitCodes.CANT_CREATE, "Could not create auto configuration directory", e);
         }
 
-        // Ensure that the files created by the auto-config command MUST have the same owner as the config dir itself,
-        // as well as that the replaced files don't change ownership.
-        // This is because the files created by this command have hard-coded "no" permissions for "group" and "other"
-        UserPrincipal newFileOwner = Files.getOwner(instantAutoConfigDir, LinkOption.NOFOLLOW_LINKS);
-        if ((false == newFileOwner.equals(Files.getOwner(env.configFile(), LinkOption.NOFOLLOW_LINKS))) ||
-                (false == newFileOwner.equals(Files.getOwner(ymlPath, LinkOption.NOFOLLOW_LINKS))) ||
-                (Files.exists(keystorePath) && false == newFileOwner.equals(Files.getOwner(keystorePath, LinkOption.NOFOLLOW_LINKS)))) {
-            Files.deleteIfExists(instantAutoConfigDir);
-            if (options.has(strictOption)) {
-                throw new UserException(ExitCodes.CONFIG, "Aborting auto configuration because it would change config file owners");
-            } else {
-                return; // if a different user runs ES compared to the user that installed it, auto configuration will not run
-            }
+        // Check that the created auto-config dir has the same owner as the config dir.
+        // This is a sort of sanity check.
+        // If the node process works OK given the owner of the config dir, it should also tolerate the auto-created config dir,
+        // provided that they both have the same owner and permissions.
+        final UserPrincipal newFileOwner = Files.getOwner(instantAutoConfigDir, LinkOption.NOFOLLOW_LINKS);
+        if (false == newFileOwner.equals(Files.getOwner(env.configFile(), LinkOption.NOFOLLOW_LINKS))) {
+            deleteDirectory(instantAutoConfigDir);
+            // the following is only printed once, if the node starts successfully
+            throw new UserException(ExitCodes.CONFIG, "Aborting auto configuration because of config dir ownership mismatch");
         }
+        final KeyPair transportKeyPair;
+        final X509Certificate transportCert;
+        final KeyPair httpCAKeyPair;
+        final X509Certificate httpCACert;
+        final KeyPair httpKeyPair;
+        final X509Certificate httpCert;
+        try {
+            // the transport key-pair is the same across the cluster and is trusted without hostname verification (it is self-signed),
+            final X500Principal certificatePrincipal = new X500Principal(buildDnFromDomain(System.getenv("HOSTNAME")));
+            final X500Principal caPrincipal = new X500Principal(AUTO_CONFIG_ALT_DN);
+            // this does DNS resolve and could block
+            final GeneralNames subjectAltNames = getSubjectAltNames();
```

Comment:
I know I coded this, but I now have second thoughts.

(Reverse) DNS for all the IPs on all network interfaces has the potential to seriously slow down node start-up.
Moreover, including the DNS names inside the HTTP cert is not immediately useful because the Kibana instances are initially going to be configured with IP addresses for the ES nodes (because this is what we include in the enrollment token, also because the ES node doesn't do DNS resolving itself) - but it is true that a cert with DNS names is more useful (though using it means changing the Kibana configuration, to use the DNS name of the ES nodes).

I think at the very least we should time bound it, to eg 1 sec (and start it async).
But maybe we're OK, for now, to not include DNS names in the http cert? It's annoying because this way we're not resilient to nodes changing the IP addresses. We can later discuss, including DNS names in enrollment tokens and certificates, WDYT?

Reply:
> how about we limit this to :

    IP Addresses
    localhost
    $HOSTNAME

for now and take it from there in the future if needed ?

Sounds good to me.